python-api/livy/client.py

```diff
@@ -0,0 +1,167 @@
+#
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from ConfigParser import SafeConfigParser
+from StringIO import StringIO
+import os
+import codecs
+
+
+class HttpClient(object):
+    """A http based client for submitting Spark-based jobs to a Livy backend."""
+
+    __CONFIG_SECTION = 'env'
+    __LIVY_CLIENT_CONF_DIR = "LIVY_CLIENT_CONF_DIR"
```

Comment:
Except in exceedingly rare circumstances, I recommend never using two preceding underscores on any member variables or functions.